### PR TITLE
Introduce cron-like scheduling for producers.

### PR DESCRIPTION
**Pull request by ralphbean**
This fixes one of our major underlying issues raised in #6. With this,
we can have our producers fire at specific times.

### CODE REVIEW

**Comment by lmacken** [comment]
Real nice. :+1:

File: tests/test_schedule.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @freezegun.freeze_time('2012-01-14 00:00:00')
+    def test_working_with_time_sleep(self):
+        s = Schedule(second=[1])
+        time.sleep(s)  # Let's just make sure this doesn't crash
+
+    @nose.tools.raises(ValueError)
+    def test_wrap_invalid_item(self):
+        Schedule(minute=['lol'])
+
+    @nose.tools.raises(TypeError)
```
Comment (pypingou) [inline tests/test_schedule.py:57]: This is neat! :)

File: statscache/schedule.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+class Minute(Denomination):
+    default = range(0, 60)
+
+
+class Hour(Denomination):
+    default = range(0, 24)
+
+
+class Schedule(object):
```
Comment (rtnpro) [inline statscache/schedule.py:38]: @ralphbean Hats off! :+1:

**Comment by rtnpro** [comment]
:+1: Landing it on develop.

**Comment by rtnpro** [comment]
@ralphbean Test case are failing at my end.

**Pull request merged**